metaflow/plugins/kfp/kfp.py

```diff
@@ -1244,9 +1042,18 @@ def _create_exit_handler_op(self) -> ContainerOp:
         if self.notify_on_success:
             notify_variables["METAFLOW_NOTIFY_ON_SUCCESS"] = self.notify_on_success
 
-        return exit_handler(
-            flow_name=self.name,
-            status="{{workflow.status}}",
-            kfp_run_id=dsl.RUN_ID_PLACEHOLDER,
-            notify_variables=notify_variables,
-        )
+        exit_handler_command = [
+            "bash",
```

Comment:
Do you need the `bash -ec` here?

Reply:
It's used to chain bash commands together - for some reason, it looks like `container_op` only allows a single bash command with `bash -ec`.

Reply:
Yeah that's really a Docker issue, it needs to start with a single process and all `bash` really does unsurprisingly is launch processes so it's a great choice in this case. I think your answer on the `init_cmd` answered my question that we chain many commands together, I do wonder if we can move these into `python` and then we'd have a cleaned spec but definitely not a major blocker so resolving this to unblock you 😄

Reply:
Alex's explanation is a good comment to add to the code.